### ISSUE TITLE
pycodestyle is not compatible with `PYTHONOPTIMIZE=2`

### ISSUE THREAD

**Issue by mondeja** [PyCQA/pydocstyle#606]
pycodestyle [relies on function doctrings](https://github.com/PyCQA/pydocstyle/blob/a6fe4222012e990042c86cdaaa904a8d059752ee/src/pydocstyle/checker.py#L167) to show the linting errors. When you run a program with `PYTHONOPTIMIZE=2` environment variable or [the `-OO` option](https://docs.python.org/3/using/cmdline.html#cmdoption-OO) the docstrings are stripped, so the next error is raised:

```
... (traceback)

  File "/.../site-packages/pydocstyle/checker.py", line 160, in check_source
    partition = this_check.__doc__.partition('.\n')
AttributeError: 'NoneType' object has no attribute 'partition'
```
